blog: add StrykerJS 5 article

nicojs opened the pull request.

hugo-vrijswijk commented on blog/2021-05-15-announcing-stryker-5-brave-new-dimension.md:
```diff
@@ -0,0 +1,141 @@
+
+If you're new to mutation testing, it's a way to measure your tests' effectiveness. A mutation testing framework will make small changes, called _mutants_, one by one in your source code. Then it will run your tests to see if one of them fails. If so, you just "killed" that mutant; if not, it "survived". If too many mutants survive, you probably want to improve your tests. The mutation testing re ...
+
+If you're already using Stryker, you can install this latest version with your package manager of choice.
```
Why do I need to install if I'm already using Stryker? Wouldn't I need to update instead?

And shouldn't it be StrykerJS 😬

nicojs replied:
I guess 🙄


😉

nicojs replied:
I'll change it to `npm i -D @stryker-mutator/core@latest`

hugo-vrijswijk commented on blog/2021-05-15-announcing-stryker-5-brave-new-dimension.md:
````diff
@@ -0,0 +1,141 @@
+  -export const hello ='👋'
+  +export const hello = ''
+  ```
+  Tests are never able to cover such mutants (yet they can still kill them).
````
This confused me. A test never covers a mutant but can still kill it? What is this magic

nicojs replied:
How would you describe it?

hugo-vrijswijk replied:
I'm not sure what you mean by it never covering such mutants but still being able to kill them, so I don't know how I'd describe it 😅. Are you talking about static mutants?

hugo-vrijswijk commented on blog/2021-05-15-announcing-stryker-5-brave-new-dimension.md:
```diff
@@ -0,0 +1,141 @@
+
+StrykerJS typically doesn't mutate your files directly. Instead, it creates a copy inside a _sandbox_ directory, usually located at `.stryker-tmp/sandbox-123456`. However, it won't copy over all files there to save time. Stryker used to rely on git to decide which files to copy. Any `.gitignored` files would not be copied.
+
+There were some issues with this approach. First of all, it required your project to be located in a git repository and you needed to have git installed. And then, there are also examples of test suites that need some `.gitignored` files to run.
```
```suggestion
There were some issues with this approach. First of all, it required your project to be located in a git repository and you needed to have git installed. And then, there are also examples of test suites that need some `.gitignore`d files to run.
```

hugo-vrijswijk commented on blog/2021-05-15-announcing-stryker-5-brave-new-dimension.md:
```diff
@@ -0,0 +1,141 @@
+
+## 🕵️‍♂️ Default coverage analysis
+
+Coverage analysis is a feature that makes StrykerJS run faster because it will run fewer tests per mutant. If you are to enable this feature depends on your test suite and test runner, see [coverage analysis in configuration](https://stryker-mutator.io/docs/stryker-js/configuration#coverageanalysis-string) for more details.
```
The flow of this paragraph is a bit weird, I'm not quite sure what you mean

hugo-vrijswijk commented on blog/2021-05-15-announcing-stryker-5-brave-new-dimension.md:
````diff
@@ -0,0 +1,141 @@
+
+  ```json
+  {
+    "ignorePatterns": ["**", "!foo.js"]
````
Should this be higher with the change explanation to show people how to migrate their config easier?

nicojs commented:
Thanks for the review @hugo-vrijswijk . I've implemented your suggestions and improved some more stuff. Also added a small section about mutating `/regex/`s. 

If you approve, I want to deploy this tonight

hugo-vrijswijk reviewed (approved):
Ship it! 🚀

nicojs commented:
👨‍🚀 https://stryker-mutator.io/blog/announcing-stryker-js-5-brave-new-dimension.md 🤘
